src/Network/Network/Common/NetworkResourceManagerProfile.cs

```diff
@@ -89,6 +89,37 @@ public override string ProfileName
             }
         }
 
+        public class VirtualRouterConverter : ITypeConverter<MNM.VirtualRouter, CNM.PSVirtualRouter>
+        {
+            public CNM.PSVirtualRouter Convert(MNM.VirtualRouter source, CNM.PSVirtualRouter destination, ResolutionContext context)
+            {
+                var router = new CNM.PSVirtualRouter()
+                {
+                    Id = source.Id,
+                    Name = source.Name,
+                    Etag = source.Etag,
+                    Location = source.Location,
+                    Type = source.Type,
+                    ProvisioningState = source.ProvisioningState,
+                    VirtualRouterAsn = (uint) source.VirtualRouterAsn
+                };
+                if (source.HostedGateway != null)
+                {
+                    router.HostedGateway = new CNM.PSResourceId()
+                    {
+                        Id = source.HostedGateway.Id
+                    };
+                }
+                if (source.VirtualRouterIps != null)
+                {
+                    router.VirtualRouterIps = source.VirtualRouterIps.ToList<string>();
+                }
+
+                router.Peerings = new List<CNM.PSVirtualRouterPeer>();
```

Comment:
You sure you don't need to map all peerings? I get that you with poll for them in Get, but there may be other cases where this model is going to be used

Reply:
I would prefer to fetch all peerings for the virtualrouter. I totally see where you are coming from. For now, I don't believe populating peerings here is needed